### DISCUSSION TITLE
add volume up and volume down option

### DISCUSSION THREAD

**Discussion by ryan00davis** [mrene/minidsp-rs#225] [Ideas]
Any chance you could add an option to run as minidsp.exe -vol-up <optional db step>?
That way, i could link it to a keyboard shortcut or a dial of some kind to step the volume up and down (either by the default amount, or by a set db step). Having to set the specific volume level makes it not easy to work with incrementally.

**Comment by mrene** [comment]
That can be done, but only for the master volume, since it's not possible to read individual gain values for inputs and outputs. (Added #230 to track this)

**Reply by ryan00davis** [reply]
That would solve all of my problems, master volume is the only one I'd care to adjust this way. Thank you.

**Reply by mrene** [reply]
I added a relative option to the `gain` subcommand. If you want to increase by 5dB you can do `minidsp gain -r 5` or `minidsp gain --relative 5`.

To decrease, you have to use -- to separate the options from the values. `minidsp gain -r -- -5`

**Reply by mglobetx** [reply]
Is it possible to add this to the API also?

**Reply by suicidaleggroll** [reply]
I second the request to get this added to the api, if possible.

**Reply by mrene** [reply]
This was implemented in https://github.com/mrene/minidsp-rs/pull/722
